rudder-agent/SOURCES/check-rudder-agent

```diff
@@ -134,10 +131,10 @@ check_and_fix_cfengine_processes() {
 
   # Check for anomalous number of CFEngine processes
   # If there are more than 8 agent/executor processes, we should kill them, and purge the lock database
-  if [ ${NB_CF_PROCESS_RUNNING} -gt 8 ]; then
+  if [ "${NB_CF_PROCESS_RUNNING}" -gt 8 ]; then
     echo_n "WARNING: Too many instance of CFEngine processes running. Killing them, and purging the CFEngine lock database..."
     echo "${CF_PROCESS_RUNNING}" | awk 'BEGIN { OFS=" "} {print $2 }' | xargs kill -9 || true
-    if [ "z${OS_FAMILY}" != "zAIX" ]; then
+    if [ -e /etc/init.d/rudder-agent ] && [ "${OS_FAMILY}" != "AIX" ] || [ "${OS_FAMILY}" = "SunOS" ]; then
```

Comment:
The z is inherited from the past, when people didn't use double quote around variables, they had to put a letter to avoid empty variables